aws/resource_aws_ecs_task_set.go

```diff
@@ -0,0 +1,731 @@
+					if err != nil {
+						errors = append(errors, fmt.Errorf(
+							"%q cannot be parsed as a duration: %s", k, err))
+					}
+					if duration < 0 {
+						errors = append(errors, fmt.Errorf(
+							"%q must be greater than zero", k))
+					}
+					return
+				},
+			},
+
+			"tags": tagsSchema(),
+		},
+	}
+}
+
+func resourceAwsEcsTaskSetCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ecsconn
+
+	cluster := d.Get("cluster").(string)
+	service := d.Get("service").(string)
+	input := ecs.CreateTaskSetInput{
+		ClientToken:    aws.String(resource.UniqueId()),
+		Cluster:        aws.String(cluster),
+		Service:        aws.String(service),
+		TaskDefinition: aws.String(d.Get("task_definition").(string)),
+		Tags:           keyvaluetags.New(d.Get("tags").(map[string]interface{})).IgnoreAws().EcsTags(),
+	}
+
+	if v, ok := d.GetOk("external_id"); ok {
+		input.ExternalId = aws.String(v.(string))
+	}
+
+	if v, ok := d.GetOk("launch_type"); ok {
+		input.LaunchType = aws.String(v.(string))
+	}
+
+	input.CapacityProviderStrategy = expandEcsCapacityProviderStrategy(d.Get("capacity_provider_strategy").(*schema.Set))
+
+	loadBalancers := expandEcsLoadBalancers(d.Get("load_balancer").([]interface{}))
+	if len(loadBalancers) > 0 {
+		log.Printf("[DEBUG] Adding ECS load balancers: %s", loadBalancers)
+		input.LoadBalancers = loadBalancers
+	}
+
+	input.NetworkConfiguration = expandEcsNetworkConfiguration(d.Get("network_configuration").([]interface{}))
+
+	if v, ok := d.GetOk("platform_version"); ok {
+		input.PlatformVersion = aws.String(v.(string))
+	}
+
+	scale := d.Get("scale").([]interface{})
+	if len(scale) > 0 {
+		input.Scale = expandAwsEcsScale(scale[0].(map[string]interface{}))
+	}
+
+	serviceRegistries := d.Get("service_registries").([]interface{})
+	if len(serviceRegistries) > 0 {
+		input.ServiceRegistries = expandAwsEcsServiceRegistries(serviceRegistries)
+	}
+
+	log.Printf("[DEBUG] Creating ECS Task set: %s", input)
+
+	// Retry due to AWS IAM & ECS eventual consistency
+	var out *ecs.CreateTaskSetOutput
+	var err error
+	err = resource.Retry(2*time.Minute, func() *resource.RetryError {
+		out, err = conn.CreateTaskSet(&input)
+
+		if err != nil {
+			if isAWSErr(err, ecs.ErrCodeClusterNotFoundException, "") ||
+				isAWSErr(err, ecs.ErrCodeServiceNotFoundException, "") ||
+				isAWSErr(err, ecs.ErrCodeTaskSetNotFoundException, "") ||
+				isAWSErr(err, ecs.ErrCodeInvalidParameterException, "does not have an associated load balancer") {
+				return resource.RetryableError(err)
+			}
+			return resource.NonRetryableError(err)
+		}
+
+		return nil
+	})
+
+	if isResourceTimeoutError(err) {
+		out, err = conn.CreateTaskSet(&input)
+	}
+
+	if err != nil {
+		return fmt.Errorf("Error creating ECS TaskSet: %s", err)
+	}
+
+	taskSet := *out.TaskSet
+
+	log.Printf("[DEBUG] ECS Task set created: %s", aws.StringValue(taskSet.TaskSetArn))
+	d.SetId(aws.StringValue(taskSet.TaskSetArn))
```

Comment:
can you please also add a computed attribute `arn` with this value in the read function,
see https://github.com/terraform-providers/terraform-provider-aws/blob/master/docs/contributing/running-and-writing-acceptance-tests.md#basic-acceptance-tests for the test also needed for this.